src/compiler/reviews.md

```diff
@@ -1,4 +1,156 @@
+    > and if they are impartial enough to provide a sufficiently unbiased perspective.
+
+
+
+## Reviewing Checklist
+
+The following list of questions (to be posted as part highfive bot's automatic PR response)
+will help reviewers and PR authors alike to bring PRs into good shape and meet the above criteria:
+
+> #### Checklist for PR authors and reviewers
+> * Does the PR message have
+>    * a concise **high-level description** of the changes? (*what* is being changed)
+>    * a clear **rationale** for doing them? (*why* is it being changed)
+>    * a list of potential **points of contention**?
+>    * **links to relevant issues**, RFCs, MCPs, etc?
+> * Does the PR need a **regression test**?
+> * Does the change need to be covered by a **[major change proposal][mcp]**? Is it already covered?
+> * Would someone trying to understand the PR in a year's time be able to quickly **reconstruct what's going on**?
+> * Is the new code **properly documented**? Is existing documentation still up-to-date?
+> * Does the PR introduce a **regression** of any of the following:
+>   * error message quality
+>   * maintainability (e.g. complex code, no documentation, unsafe)
+>   * any specific target platforms
+>   * downstream tooling (e.g. linkers, debuggers)
+>   * compile times
+>   * memory usage
+>
+> #### Checklist for reviewers
+> * Am I the **right person to review** this PR:
+>   * Do I understand the code well enough?
+>   * Would I be able to spot non-obvious side effects?
+>   * Would I be able to fix a bug introduced by this PR?
+>   * Can I do the review in a timely fashion?
+>   * Do I feel pressure to quickly approve the PR for some reason?
+>   * Am I impartial enough?
+> * Before merging:
+>   * Is the **PR message still accurate**?
+>   * Is the **commit history clean** enough?
+
+
+
+## Guidance for dealing with common situations
+
+In most cases common sense is enough for deciding how to apply this policy.
+However, sometimes there are gray areas where it is not immediately clear how to proceed.
+This section lists a few common cases together with guidance on how to deal with them.
+
+* ### I don't think I am a good fit for reviewing - what now?
+  It is completely normal that you get assigned a PR (via highfive or otherwise) but don't feel comfortable reviewing it.
+  Here is what you can do, depending on the concrete case:
+
+  - If the change seems really big or contentious, consider asking for an MCP (see below).
+  - If you know just the right person for the review, assign them via `r? @<github-name>`.
+    It's polite to leave a comment asking them if they can take over --
+    but you don't have to make sure beforehand that they can actually do it.
+  - If you don't know the code well or already have too much on your plate,
+    ask highfive to roll the dice again via `r? @rust-lang/compiler-contributors`.
+
+  You can also always ask specific compiler team members for help finding a reviewer.
```

Comment:
I think this will probably come up with the discussion on getting more people in the review queue, but I think we should also emphasize that even if you can't do the *final* review for whatever reason, it's always okay and *helpful* to take a first glance and give comments. It's a win-win all around: contributors might get at least some feedback sooner; the initial reviewer can at least gets eyes on code that they might be new to; the final reviewer might have less things that need to be said/addressed when they get to their review.

Reply:
+1!

Reply:
Also, can we dedicate a Zulip stream for this? That seems more approachable than asking specific compiler team members (which member to ask?)

Reply:
A Zulip stream sounds like a good idea. Any suggestions for the name of such a stream?

Reply:
`#t-compiler/reviews` perhaps?

Reply:
I switched this to the Zulip stream version.